src/main/java/uk/gov/homeoffice/digital/sas/balancecalculator/BalanceCalculator.java

```diff
@@ -107,10 +114,11 @@ Agreement getAgreementApplicableToTimeEntryEndDate(String tenantId,
   }
 
   SortedMap<AccrualType, SortedMap<LocalDate, Accrual>> getImpactedAccruals(
-      String tenantId, String personId, LocalDate startDate, LocalDate endDate) {
+      String tenantId, String personId, String timeEntryId,
+      LocalDate timeEntryStartDate, LocalDate agreementEndDate) {
 
-    List<Accrual> accruals = accrualsService.getImpactedAccruals(tenantId, personId,
-        startDate, endDate);
+    List<Accrual> accruals = accrualsService.getImpactedAccruals(
+        tenantId, personId, timeEntryId, timeEntryStartDate, agreementEndDate);
```

Comment:
Should this not be timeEntryEndDate instead of agreementEndDate

Reply:
Correct, change to timeEntryEndDate